--- a/pkgs/by-name/ar/arcan/package.nix
+++ b/pkgs/by-name/ar/arcan/package.nix
@@ -51,0 +51,1 @@
+  debug ? false,
@@ -60,0 +61,1 @@
+  storePath ? "$out"
@@ -120,0 +122,1 @@
+    (lib.cmakeFeature "CMAKE_BUILD_TYPE" (if debug then "Debug" else "Release"))
@@ -168,2 +171,2 @@
-      --replace-fail "/usr/bin" "$out/bin" \
-      --replace-fail "/usr/share" "$out/share"
+      --replace-fail "/usr/bin" "${storePath}/bin" \
+      --replace-fail "/usr/share" "${storePath}/share"
