--- a/pkgs/by-name/ar/arcan/package.nix
+++ b/pkgs/by-name/ar/arcan/package.nix
@@ -50,0 +50,1 @@
+  debug ? false,
@@ -59,0 +60,1 @@
+  storePath ? "$out"
@@ -119,0 +121,1 @@
+    (lib.cmakeFeature "CMAKE_BUILD_TYPE" (if debug then "Debug" else "Release"))
@@ -167,2 +170,2 @@
-      --replace-fail "/usr/bin" "$out/bin" \
-      --replace-fail "/usr/share" "$out/share"
+      --replace-fail "/usr/bin" "${storePath}/bin" \
+      --replace-fail "/usr/share" "${storePath}/share"
