--- a/pkgs/by-name/ar/arcan/wrapper.nix
+++ b/pkgs/by-name/ar/arcan/wrapper.nix
@@ -2,0 +3,5 @@
+  doas,
+  fossil,
+  gdb,
+  inotify-tools,
+  lldb,
@@ -3,0 +9 @@
+  nvim-arcan,
@@ -4,0 +11 @@
+  wpa_supplicant,
@@ -18,0 +26,7 @@
+        --prefix PATH ":" "${inotify-tools}/bin" \
+        --prefix PATH ":" "${doas}/bin" \
+        --prefix PATH ":" "${fossil}/bin" \
+        --prefix PATH ":" "${gdb}/bin" \
+        --prefix PATH ":" "${lldb}/bin" \
+        --prefix PATH ":" "${wpa_supplicant}/bin" \
+        --prefix PATH ":" "${nvim-arcan}/bin" \
@@ -19,0 +34,1 @@
+        --set LASH_BASE "${placeholder "out"}/share/arcan/lash" \
